databases/business-sync/fragment-shoebox-snippet.js

```diff
@@ -0,0 +1,40 @@
+{
+  channels: getDocSyncChannels(doc, oldDoc),
+  typeFilter: function(doc, oldDoc) {
+    return createBusinessEntityRegex('shoeboxSnippet\\.[A-Za-z0-9_-]+\\.[A-Za-z0-9_-]+').test(doc._id);
+  },
+  allowUnknownProperties: true,
+  immutable: true,
+  propertyValidators: {
+    type: {
+      // The type of shoebox snippet (ie, bank transaction, email, photo, digital receipt)
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    source: {
+      // The source of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    kashooEntityType: {
+      // Type of corresponding Books entity
+      type: 'string',
+      required: false,
+      mustNotBeEmpty: true
+    },
+    kashooId: {
+      // The ID of the corresponding Books entity
+      type: 'integer',
+      required: false,
+      minimumValue: 1
+    },
+    data: {
+      // Raw data of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    }
```

Comment:
Definitely a 👍 on a timestamp. Do we have an agreed convention (e.g. `created`, `createdAt`, `creationTime`)? In this instance I think there is a creation/received timestamp in our system that is definitely different than the transaction time (when the transaction was booked in Square).

Updates are a good question. I think I'm tending to defer on this one for now although my gut says there might be an array of updates, each of which contains just the fields of the document that have deltas and the new value. That is probably a pre-mature optimization as opposed to having a whole document. And each would also have a receivedAt (or agreed name from previous section) time to know when the update was received.

We are going to need it. Square has confirmed that already. But let's take a KISS/YAGNI attitude and come back to it.